lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -30,19 +27,54 @@ import (
 	lte_protos "magma/lte/cloud/go/protos"
 	"magma/lte/cloud/go/serdes"
 	lte_models "magma/lte/cloud/go/services/lte/obsidian/models"
-	"magma/lte/cloud/go/services/subscriberdb/obsidian/models"
+	"magma/lte/cloud/go/services/subscriberdb/storage"
 	"magma/orc8r/cloud/go/services/configurator"
 	"magma/orc8r/lib/go/protos"
 )
 
 type subscriberdbServicer struct {
 	flatDigestEnabled bool
+	useDigestStore    bool
+	digestStore       storage.DigestLookup
 }
 
-const defaultSubProfile = "default"
+func NewSubscriberdbServicer(config subscriberdb.Config, digestStore storage.DigestLookup) lte_protos.SubscriberDBCloudServer {
+	return &subscriberdbServicer{
+		flatDigestEnabled: config.FlatDigestEnabled,
+		useDigestStore:    config.UseDigestStore,
+		digestStore:       digestStore,
+	}
+}
+
+// QueryFlatDigest checks if the client-side subscriber digest is up-to-date with the
+// cloud version, and returns signal for whether the client needs an update.
+func (s *subscriberdbServicer) QueryFlatDigest(ctx context.Context, req *lte_protos.QueryFlatDigestRequest) (*lte_protos.QueryFlatDigestResponse, error) {
+	gateway := protos.GetClientGateway(ctx)
+	if gateway == nil {
+		return nil, status.Errorf(codes.PermissionDenied, "missing gateway identity")
+	}
+	if !gateway.Registered() {
+		return nil, status.Errorf(codes.PermissionDenied, "gateway is not registered")
+	}
+	networkID := gateway.NetworkId
+	clientDigest := req.Digest
+	cloudDigest, noUpdates := s.getDigestInfo(clientDigest, networkID)
 
-func NewSubscriberdbServicer(config subscriberdb.Config) lte_protos.SubscriberDBCloudServer {
-	return &subscriberdbServicer{flatDigestEnabled: config.FlatDigestEnabled}
+	return &lte_protos.QueryFlatDigestResponse{
```

Comment:
One-line returns are preferable for readability -- so create the return value and then call `return` on separate lines